### PR TITLE
More wiring and documentation around Macaw

### LINKED ISSUE #43096: Provide Macaw with driver-based formatting options
In order to [correctly handle case and quoting during SQL rewrites](https://github.com/metabase/macaw/issues/34), we need to be driver-aware.

As a first version we'll expect Metabase to pass in some relevant config, and keep Macaw itself driver agnostic.
```clojure
{:case-insensitive? true
 :quotes            "[]" ;; MSSQL
```

Perhaps we're able to leverage existing driver methods used by the query processor, especially for the quotes. Otherwise, new multimethods with sensible defaults (thinking case sensitive, with double quotes for quotes).

In future we might find the need to have dialect specific grammars, in which case this will seem a bit redundant. We'll probably keep a fairly agnostic superset grammar as fallback for newfangled 3rd party drivers, which will still take these options though.

### PR DESCRIPTION
**Pull request by crisptrutski**
Closes #43096 

### Description

This wires in the new Macaw options in the final spot, which isn't actually used by the application yet.

### CODE REVIEW

File: src/metabase/native_query_analyzer/impl.cljc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+(ns metabase.native-query-analyzer.impl)
```
Comment (crisptrutski) [inline src/metabase/native_query_analyzer/impl.cljc:1]: No changes here, it was just moved to be reused between namespaces.

**Comment by crisptrutski** [comment]
This has surfaced a Macaw bug, which is that it expects the `replacements` to be in lowercase if we're using case insensitivity.

File: src/metabase/native_query_analyzer/replacement.clj [hunk 2]
```diff
@@ -141,11 +142,18 @@
...
   with the appropriate replacements made."
-  [query renames]
-  (let [raw-query                    (get-in query [:native :query])
-        parsed-query                 (params.parse/parse raw-query)
-        param->value                 (param-values query)
-        {clean-query :query
-         tt-subs     :substitutions} (parse-tree->clean-query raw-query parsed-query param->value)
-        renamed-query                (macaw/replace-names clean-query renames {:allow-unused? true})]
-    (replace-all renamed-query tt-subs)))
+  ;; This arity exists as a convenient for all the tests that are fairly driver agnostic.
```
Comment (tsmacdonald) [inline src/metabase/native_query_analyzer/replacement.clj:145]:
```suggestion
  ;; This arity exists as a convenience for all the tests that are fairly driver agnostic.
```

File: src/metabase/native_query_analyzer/impl.cljc [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+;; NOTE: In future we would want to replace [[considered-drivers]] and [[macaw-options]] with (a) driver method(s),
```
Comment (tsmacdonald) [inline src/metabase/native_query_analyzer/impl.cljc:3]:
```suggestion
;; NOTE: In the future we would want to replace [[considered-drivers]] and [[macaw-options]] with (a) driver method(s),
```

**Review by tsmacdonald — APPROVED** [review]

**Comment by crisptrutski** [comment]
This will be unblocked by https://github.com/metabase/macaw/pull/48

**Pull request merged**